cpp/src/io/json/read_json.cu

```diff
@@ -253,14 +261,30 @@ datasource::owning_buffer<rmm::device_buffer> get_record_range_raw_input(
   std::int64_t buffer_offset = 0;
   auto readbufspan =
     ingest_raw_input(bufspan, sources, chunk_offset, chunk_size, delimiter, stream);
+  auto const requested_size = readbufspan.size();
 
   auto const shift_for_nonzero_offset = std::min<std::int64_t>(chunk_offset, 1);
   auto const first_delim_pos =
     chunk_offset == 0 ? 0 : find_first_delimiter(readbufspan, delimiter, stream);
+
+  // If we read till the end of the last source, we cannot be sure
+  // if the last record read ends with a delimiter. In such cases, we add a delimiter
+  // nevertheless; even if the record terminates
+  // with a delimiter, adding a extra delimiter does not affect the table constructed since the
+  // parser ignores empty lines.
+  auto insert_delimiter = [delimiter, stream](device_span<char> subspan) {
```

Comment:
Interesting, to use a `device_span` for copying essentially a single byte.  I would have thought a pointer would have done.  
On the other hand, that the argument is a device address is more apparent here.